src/NuGet.Core/NuGet.Protocol/RemoteRepositories/HttpFileSystemBasedFindPackageByIdResourceProvider.cs

```diff
@@ -23,6 +25,15 @@ public HttpFileSystemBasedFindPackageByIdResourceProvider()
             var serviceIndexResource = await sourceRepository.GetResourceAsync<ServiceIndexResourceV3>(token);
             var packageBaseAddress = serviceIndexResource?.GetServiceEntryUris(ServiceTypes.PackageBaseAddress);
 
+            foreach (Uri endpoint in packageBaseAddress)
+            {
+                PackageSource serviceSource = new PackageSource(endpoint.ToString(), endpoint.ToString());
```

Comment:
We need to minimize memory allocations to minimize performance slowdowns. Uri has a property `OriginalString`, which I expect can be used and shouldn't allocate.